src/ml_tooling/transformers/rare_feature.py

```diff
@@ -0,0 +1,37 @@
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+class RareFeatureEncoder(BaseEstimator, TransformerMixin):
+    """
+    Replace values which occurs below the chosen threshold.
+    The threshold can be a procent or int value.
+    The values are replaced with the chosen fill value.
+    Column names need to be the same.
+    """
+
+    def __init__(self, threshold=0.2, fill_rare="Rare"):
+        self.threshold = threshold
+        self.fill_rare = fill_rare
+        self.mask_dict = {}
+
+    def fit(self, X: pd.DataFrame, y=None):
+        for col in X.columns:
+            if self.threshold < 1:
+                frequencies = X[col].value_counts(normalize=True)
+            else:
+                frequencies = X[col].value_counts()
+
+            condition = frequencies < self.threshold
+            mask_obs = frequencies[condition].index
```

Comment:
Would probably combine these two, since we don't refer to it afterwards

```suggestion
            mask_obs = frequencies[frequencies < self.threhold].index
```